src/lib/learn-client/api/progress/get-all-progress.ts

```diff
@@ -0,0 +1,56 @@
+import queryString from 'query-string'
+import { get, toError } from 'lib/learn-client'
+import { Tutorial, Collection } from 'lib/learn-client/types'
+import { SessionData } from 'types/auth'
+import { PROGRESS_API_ROUTE } from '.'
+import { ApiCollectionTutorialProgress } from '../api-types'
+
+interface GetAllProgressOptions {
+	accessToken: SessionData['accessToken']
+	tutorialIds?: Tutorial['id'][]
+	collectionIds?: Collection['id'][]
+}
+
+type GetAllProgressResult = ApiCollectionTutorialProgress[]
+
+/**
+ * Fetches all tutorial progress records for the current user.
+ * If successful, returns an array of progress objects.
+ * Otherwise throws an error.
+ *
+ * Accepts optional arrays of `tutorialIds` and `collectionIds`.
+ * When either of these are provided, we return all progress records
+ * that match either:
+ * - any one of the provided `tutorialIds`
+ * - any one of the provided `collectionIds`
+ *
+ * https://digital-api-specs.vercel.app/learn#tag/Progress/paths/~1progress/get
+ */
+const getAllProgress = async ({
```

Comment:
This does the API data fetching we need for `use-collection-progress`.

It'll also be used later for batch fetching tutorial progress to prime individual queries.

> **Note**: We could use the [collection-specific endpoint](https://digital-api-specs.vercel.app/learn#tag/Collection-Tutorial-Progress/paths/~1collections~1{id}~1progress/get) instead... but with `collectionId` passed, and used as a query param, this endpoint does the exact same thing, and will definitely be used for things like batch fetching tutorial progress for the sidebar. So figured it made sense to set up `get-all-progress` in `learn-client` first (heck, we may not even need a separate `get-collection-progress`? since `use-collection-progress` can always use `get-all-progress` under the hood).